### PR TITLE
Add changes necessary for addition of Error constructor to status types

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
This is labeled do not merge, because it will not compile against most recent release of devicemapper.

**Comment by mulkieran** [comment]
Note that this PR in a sense compounds the following existing problems:
* https://github.com/stratis-storage/stratisd/issues/1356. I do not choose to add an additional state corresponding with the Error return, because the Error return doesn't mean any state for the pool. But our choice of pool states is already very dubious.
* https://github.com/stratis-storage/stratisd/issues/1267 I added more lines to ```total_physical_used```. It continues to produce a potentially bogus value, regardless of these new lines, obviously.
* https://github.com/stratis-storage/stratisd/issues/1255 It doesn't improve the logging to give the log reader an accurate idea of the state of the pool/thinpool.
* https://github.com/stratis-storage/stratisd/issues/1243 It does not change the value of the units returned from any size methods (and then exported on the D-Bus). See also #1218.
* https://github.com/stratis-storage/stratisd/issues/1215 It adds another of many allready existing possible error returns to the check method.
* https://github.com/stratis-storage/stratisd/issues/1148 It does not make D-Bus properties that could fail safe. It could even be said to make them slightly more likely to fail, by adding another possible source of an error return.
* https://github.com/stratis-storage/stratisd/issues/1148 It doesn't fix log messages that currently aren't very useful, although it tries to make newly introduced log messages somewhat helpful.
* https://github.com/stratis-storage/stratisd/issues/512 I didn't try to improve the FilesystemStatus enum in any way. Luckily, it is not yet exported on the D-Bus.

**Comment by mulkieran** [comment]
Whoops! Trivial clippy error.

**Comment by mulkieran** [comment]
Jenkins failure was udevadm problem.

**Comment by mulkieran** [comment]
udevadm again. Reboot fedora-server-1!

**Comment by mulkieran** [comment]
Trying again.

**Comment by mulkieran** [comment]
Now trying fedora-server-2.

**Comment by mulkieran** [comment]
fedora-server-3 is also failing on something...but not everything, so maybe it's not udevadm problem.

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
Also had to update Cargo.lock.

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ authors = ["Stratis Developers <stratis-devel@lists.fedorahosted.com>"]
 dbus = {version = "0.6.4", optional = true}
 clap = "2"
 nix = "0.13"
-devicemapper = "0.26.0"
+devicemapper = {git="https://github.com/mulkieran/devicemapper-rs", branch="master-error"}
```
Comment (trgill) [inline Cargo.toml]: don't forget to remove this

**Comment by mulkieran** [comment]
rebased.

File: Cargo.toml (hunk 1 above)
Comment (tasleson) [reply Cargo.toml]: New version of devicemapper-rs is available, so you can bump the version for this PR.

**Comment by mulkieran** [comment]
Now pulling in devicemapper version 0.27.0, since release is made.

**Comment by mulkieran** [comment]
I've removed the "do not merge" flag, since there will be no conflict if this is merged.

File: src/engine/strat_engine/thinpool/filesystem.rs [hunk 2]
```diff
@@ -220,6 +220,13 @@ impl StratFilesystem {
                 // TODO: do anything when filesystem is not mounted?
                 // TODO: periodically kick off fstrim?
             }
+            ThinStatus::Error => {
+                let error_msg = format!(
+                    "Unable to get status for filesystem thin device {}",
+                    self.thin_dev.device()
+                );
```
Comment (tasleson) [inline src/engine/strat_engine/thinpool/filesystem.rs]: Would it be useful to include the fs name in this error message?  Also asking for all the other error messages that are similar in PR.

File: src/engine/strat_engine/thinpool/thinpool.rs [hunk 3]
```diff
@@ -545,6 +545,14 @@ impl ThinPool {
                 self.resume()?;
                 self.set_extend_state(data_extend_failed, meta_extend_failed);
             }
+            ThinPoolStatus::Error => {
+                // Do not set the state, because it is unknown.
+                warn!(
+                    "Devicemapper could not obtain the status for devicemapper thinpool device {} belonging to pool with UUID {}",
+                    self.thin_pool.device(),
+                    pool_uuid
+                );
```
Comment (tasleson) [inline src/engine/strat_engine/thinpool/thinpool.rs]: I believe adding the pool name would be useful for this message too.

File: src/engine/strat_engine/thinpool/filesystem.rs (hunk 2 above)
Comment (mulkieran) [reply src/engine/strat_engine/thinpool/filesystem.rs]: Filesystem name is not available here.

File: src/engine/strat_engine/thinpool/thinpool.rs (hunk 3 above)
Comment (mulkieran) [reply src/engine/strat_engine/thinpool/thinpool.rs]: Pool name is not available here.

**Pull request merged**